subsys/net/lib/nrf_cloud/src/nrf_cloud_codec.c

```diff
@@ -980,40 +984,74 @@ int nrf_cloud_obj_location_request_create(struct nrf_cloud_obj *const obj,
 		goto cleanup;
 	}
 
-	/* By default, nRF Cloud will send the location to the device */
-	if (!request_loc &&
-	    nrf_cloud_obj_num_add(&data_obj, NRF_CLOUD_LOCATION_KEY_DOREPLY, 0, false)) {
-		err = -ENOMEM;
-		goto cleanup;
-	}
-
-	switch (obj->type) {
-	case NRF_CLOUD_OBJ_TYPE_JSON:
-	{
-		/* Add cell/wifi info */
-		err = nrf_cloud_obj_location_request_payload_add(&data_obj, cells_inf, wifi_inf);
+	if (config) {
```

Comment:
if you want to avoid the change_needed flag and potential init and free of an unnecessary object:

```suggestion
	if (config && 
	    ((config->do_reply != NRF_CLOUD_LOCATION_DOREPLY_DEFAULT) || 
	     (config->hi_conf != NRF_CLOUD_LOCATION_HICONF_DEFAULT) || 
	     (config->fallback != NRF_CLOUD_LOCATION_FALLBACK_DEFAULT))) {
```

Reply:
Yes, I know, and I didn't do it that way to make the function slightly less ugly. But I'll go ahead and add it.